### PR TITLE
Add a helper function that cancels the workflow after a single output.

### PR DESCRIPTION
**Pull request by zach-klippenstein**

### CODE REVIEW

**Comment by zach-klippenstein** [comment]
TODO before merge:
- [ ] Unit tests

File: kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt [hunk 1]
```diff
@@ -109,6 +114,35 @@ fun <InputT, OutputT : Any, RenderingT, ResultT> launchWorkflowIn(
...
+ * Launches the [workflow] in a new coroutine in [scope]. The workflow tree is seeded with
+ * [initialSnapshot] and the first value emitted by [inputs].  Subsequent values emitted from
+ * [inputs] will be used to re-render the workflow.
+ *
+ * Like [launchWorkflowIn], but will automatically cancel the workflow as soon as the first
+ * [output][OutputT] is emitted. [beforeStart] gets the same [Flow] of [RenderingAndSnapshot], but
+ * instead of a flow of outputs it gets a [Deferred] of the first output.
+ */
+@UseExperimental(ExperimentalCoroutinesApi::class)
+fun <InputT, OutputT : Any, RenderingT, ResultT> launchWorkflowUntilFirstOutputIn(
```
Comment (rjrjr) [inline kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt]: Neat.

**Comment by zach-klippenstein** [comment]
Unit tests added, this is ready to go.

File: kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt [hunk 2]
```diff
@@ -115,6 +120,35 @@ fun <InputT, OutputT : Any, RenderingT, RunnerT> launchWorkflowIn(
...
```
Comment (zach-klippenstein) [inline kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt]: I kind of hate this name. `launchSingleOutputWorkflowIn`? `launchResultWorkflowIn`?

Comment (rjrjr) [reply kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt]: I like `launchSingleOutputWorkflowIn`.

giphy

**Pull request merged**